reana/reana_dev/git.py

```diff
@@ -82,6 +85,45 @@ def get_current_commit(srcdir):
     )
 
 
+def git_is_current_version_tagged(component):
+    """Determine whether the current version in source code is present as a git tag."""
+    tags_history = run_command(
+        "git tag --list", component, display=False, return_output=True
+    )
+    current_version = get_component_current_version_from_file(component)
+    return current_version in tags_history
+
+
+def git_create_release_commit(component):
+    """Create a release commit for the given component."""
+    if "release: v" in get_current_commit(get_srcdir(component)):
+        display_message("Nothing to do, last commit is a release commit.", component)
+        return False
+
+    if not git_is_current_version_tagged(component):
+        display_message(
+            f"Current version ({get_component_current_version_from_file(component)}) "
+            "not present as a git tag, plese release it and add a tag.",
+            component,
+        )
+        return False
```

Comment:
Expected to see this error message for a repo that does not contain any tags and the latest commit is not a release commit, but got instead a new `release-` branch created.

Example:

```console
$ reana-dev git-create-release-commit -c r-a-vomsproxy
[2020-08-07T16:36:21] reana-auth-vomsproxy: git checkout -b release-
Switched to a new branch 'release-'
[2020-08-07T16:36:21] reana-auth-vomsproxy: git commit -m 'release: '
On branch release-
nothing to commit, working tree clean
[2020-08-07T16:36:21] reana-auth-vomsproxy: Command 'git commit -m 'release: '' returned non-zero exit status 1.
```

Reply:
This works now, but the exit code is not set:

```console
~r/reana pr-372 reana $ reana-dev git-create-release-commit -c r-a-vomsproxy
[2020-08-11T09:52:32] reana-auth-vomsproxy: Version cannot be autodiscovered.
[2020-08-11T09:52:32] reana-auth-vomsproxy: Current version () not present as a git tag, plese release it and add a tag.

~r/reana pr-372 reana $ echo $?
0
```

Please add `sys.exit(1)` so that we fail properly if this command is plugged into a longer script.